src/compiler/diagnosticMessages.json

```diff
@@ -1928,5 +1928,9 @@
     "The 'arguments' object cannot be referenced in an arrow function. Consider using a standard function expression.": {
         "category": "Error",
         "code": 9002
+    },
+    "Tuple type '{0}' with length '{1}' cannot be assigned to tuple with length '{2}'.": {
+        "category": "Error",
+        "code": 9003

```

Comment:
Sorry, I forgot to mention this earlier - while it's less meaningful now that most checks are performed in `checker.ts`, we have codes that align with certain categories (e.g. lexical/grammatical errors, semantic errors, definition emit errors, etc.). Basically, could you change this to code 2493 and move it right after 2492.

Other than that, everything looks good to me!